### PR TITLE
Several pkg-config fixes on macOS and Windows

### PR DESCRIPTION
**Pull request by nirbheek**
f.ex. https://github.com/mesonbuild/meson/issues/4728, and many more. See commits.

### CODE REVIEW

**Comment by xclaesse** [comment]
LGTM, when CI is green.

**Comment by nirbheek** [comment]
> LGTM, when CI is green.

There's probably a lot of broken code in the dependency classes that I'll need to fix when the CI throws it at me. We were swallowing all exceptions and ignoring them when the dependency wasn't required, which was hiding programming errors like calling methods on `None`.

**Comment by z3ntu** [comment]
My project successfully compiles again with that PR: https://ci.appveyor.com/project/z3ntu/razer-test-pnxsa/build/job/6hw5o04jvrw4uas3

File: run_unittests.py [hunk 1]
```diff
@@ -230,6 +230,22 @@ def temp_filename():
...
             pass
 
+@contextmanager
+def no_pkgconfig():
+    '''
+    A context manager that overrides shutil.which to force it to return None
+    for pkg-config to simulate it not existing.
+    '''
+    old_which = shutil.which
+    def new_which(cmd, *kwargs):
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__ [E306] expected 1 blank line before a nested definition, found 0

<sub>posted by Sider</sub>

**Pull request merged**